app/src/main/kotlin/com/simplemobiletools/dialer/activities/DialpadActivity.kt

```diff
@@ -287,7 +288,14 @@ class DialpadActivity : SimpleActivity() {
         })
 
         ContactsAdapter(this, filtered, dialpad_list, null, text) {
-            startCallIntent((it as Contact).phoneNumbers.first().normalizedNumber)
+            val contact = it as Contact
+            if (config.showCallConfirmation) {
+                CallConfirmationDialog(this@DialpadActivity, contact.getNameToDisplay()) {
+                    startCallIntent(contact.phoneNumbers.first().normalizedNumber)
```

Comment:
why are you calling the first number instead of the default one? You are literally creating a bug that is fixed inside this PR in another place... And dont forget formatting the code.